cloudvolume/cloudvolume.py

```diff
@@ -160,6 +157,27 @@ def __init__(self, cloudpath, mip=0, bounded=True, autocrop=False, fill_missing=
 
     self.pid = os.getpid()
 
+  @classmethod
+  def from_numpy(cls, arr, vol_path='file://tmp/'+generate_random_string(),
+                  resolution=(4,4,40), voxel_offset=(0,0,0), 
+                  chunk_size=(64,64,64)):
+    mkdir(vol_path)
+    if arr.dtype == np.uint8 or arr.dtype == np.float32:
```

Comment:
Unfortunately, these assumptions aren't necessarily true. I think they could be good defaults, but it's important to include a layer_type option in the arguments. For example, I've seen a sparse manually reconstructed dataset with a uint8 segmentation.

Reply:
It's also important to consider uint16 and float64 for example. Integer volumes also exist.

Reply:
modified as suggested